### PR TITLE
Add PDB to teleport-kube-agent chart

### LINKED ISSUE #7330: Add a PodDisruptionBudget to Helm Charts
### What
I'd like a PodDisruptionBudget added to the Helm Charts as a definable option in teleport-cluster and probably teleport-kube-agent

### How
Add a pdb.yaml to the templates directory that would be enabled and take options from the values.yaml file.

### Why
With cluster auto scaling enabled this would ensure that there are always X pods available to service requests.

### Workaround
If a workaround exists, please include it.

I'm happy to submit a PR for this.

### PR DESCRIPTION
**Pull request by tghaas**
Mostly copied from https://github.com/gravitational/teleport/pull/7138

This does have a breaking change as I moved replicaCount under highAvailability to be consistent with the teleport-cluster chart.

Closes #7330

### CODE REVIEW

**Comment by tghaas** [comment]
@webvictim

**Comment by webvictim** [comment]
@klizhentas What do you think about the breaking change here?

**Review by webvictim — CHANGES_REQUESTED** [review]
@tghaas Thanks for the PR.

Please make `replicaCount` backwards-compatible - i.e. if `replicaCount` is set at the chart's root level, it should be treated the same as `highAvailability.replicaCount` (you can still use the new form in all docs)

We may remove this backwards compatibility in future chart versions but for now, we need to preserve the existing behaviour so running a `helm repo update` won't break functionality for existing users.

**Comment by tghaas** [comment]
@webvictim Added support for a root value of replicaCount and added a deprecation note to the values.yaml file.

**Comment by webvictim** [comment]
@tghaas Linter failed:

```
Using values from 'examples/chart/teleport-kube-agent/.lint/pdb.yaml': engine.go:167: [INFO] Missing required value: proxyAddr is required in chart values
--
172 | engine.go:167: [INFO] Missing required value: kubeClusterName is required in chart values when kube role is enabled, see README
173 | engine.go:167: [INFO] Missing required value: proxyAddr is required in chart values
174 | engine.go:167: [INFO] Missing required value: kubeClusterName is required in chart values when kube role is enabled, see README
175 | ==> Linting examples/chart/teleport-kube-agent
176 |  
177 | 1 chart(s) linted, 0 chart(s) failed
178 | Error: execution error at (teleport-kube-agent/templates/deployment.yaml:31:28): proxyAddr is required in chart values
179 |  
180 | Use --debug flag to render out invalid YAML
181 | Makefile:437: recipe for target 'lint-helm' failed
182 | make: *** [lint-helm] Error 1
```

**Comment by russjones** [comment]
@webvictim Can you take a look? If this is something we want, can you work with @tghaas to fix the conflicts. Then we can get this merged and backported to a release branch.

**Comment by tghaas** [comment]
@webvictim rebased and fixed the helm linter

File: examples/chart/teleport-kube-agent/templates/deployment.yaml [hunk 1]
```diff
@@ -35,9 +40,37 @@ spec:
       labels:
         app: {{ .Release.Name }}
     spec:
-      {{- if .Values.affinity }}
+      {{- if or .Values.affinity (gt (int .Values.highAvailability.replicaCount) 1) }}
```
Comment (webvictim) [inline examples/chart/teleport-kube-agent/templates/deployment.yaml:43]: Should this be checking against `.replicaCount` instead as that aggregates both values?

**Review by webvictim — APPROVED** [review]
LGTM - just one comment.

File: examples/chart/teleport-kube-agent/templates/deployment.yaml (hunk 1 above)
Comment (tghaas) [reply examples/chart/teleport-kube-agent/templates/deployment.yaml:43]: Yes, you're right. Let me make that change.

Do we need to update the statefulset.yaml as well? I didn't do that so we may want to.

Comment (webvictim) [reply examples/chart/teleport-kube-agent/templates/deployment.yaml:43]: The `StatefulSet` is used when persistent storage is enabled so yes, please update that one too!

Comment (tghaas) [reply examples/chart/teleport-kube-agent/templates/deployment.yaml]: @webvictim Finally back to this and will work to get it merged. Stateful set is updated as well now.

**Comment by russjones** [comment]
Will merge in https://github.com/gravitational/teleport/pull/9323.

**Pull request closed without merging**